integration_tests/CMakeLists.txt

```diff
@@ -723,7 +723,7 @@ RUN(NAME enum_06             LABELS cpython llvm c)
 RUN(NAME enum_07             IMPORT_PATH ..
                              LABELS cpython llvm c)
 RUN(NAME union_01            LABELS cpython llvm c)
-RUN(NAME union_02            LABELS cpython llvm c)
+RUN(NAME union_02            LABELS cpython llvm c NOFAST)
```

Comment:
This test silently works incorrectly on ``main`` branch as well.

Reply:
How so? How can it silently work? Is our cmake flaky? If so, that's an urgent bug to fix.

Reply:
With `main` branch,

```zsh
(lp) 14:17:37:~/lpython_project/lpython % lpython integration_tests/union_02.py --fast
0 1.00000000000000000e+00
2 7.00000000000000000e+00
5 1.30000000000000000e+01 3.04899892523928701e-314
(lp) 14:17:55:~/lpython_project/lpython % lpython integration_tests/union_02.py       
0 1.00000000000000000e+00
2 7.00000000000000000e+00
5 1.30000000000000000e+01 8.00000000000000000e+00
```

You can clearly see the difference in the last value of third line. `--fast` should have given assertion error in `main` branch but it didn't. So, clearly its passing silently in `main` branch as well. In my `libasr_sync_06` this failure just shows up as an assertion error, something which should have happened in `main` branch only. I will fix it later, after getting this PR merged.